parser-typechecker/src/Unison/Syntax/TermParser.hs

```diff
@@ -1537,19 +1572,21 @@ block' isTop implicitUnitAtEnd s openBlock closeBlock = do
             where
               step :: BlockElement v -> Term v Ann -> TermP v m
               step elem result = case elem of
-                Binding ((a, v), tm) ->
+                Binding ((a, v), tm) -> do
+                  let fullLetRecSpan = ann a <> ann result
                   pure $
                     Term.consLetRec
                       isTop
-                      (ann a <> ann result)
+                      fullLetRecSpan
                       (a, v, tm)
                       result
-                Action tm ->
+                Action tm -> do
+                  let fullLetRecSpan = (ann tm <> ann result)
                   pure $
                     Term.consLetRec
                       isTop
-                      (ann tm <> ann result)
-                      (ann tm, positionalVar (ann tm) (Var.named "_"), tm)
+                      fullLetRecSpan
+                      (Ann.External, positionalVar (ann tm) (Var.named "_"), tm)
```

Comment:
We really shouldn't be using `User` vars for generated bindings like this; but at the very least I need to mark it as External to the file.

Reply:
Agree about the `User` vars, but it's okay to fix it separately.